components/style/stylist.rs

```diff
@@ -98,6 +101,16 @@ pub struct Stylist {
 
     /// The total number of times the stylist has been rebuilt.
     num_rebuilds: usize,
+
+    /// The set of registered custom property associated with the document.
+    /// Should be initialized as soon as possible.
+    #[cfg_attr(feature = "servo", ignore_heap_size_of = "Arc")]
+    registered_property_set: Option<Arc<RwLock<RegisteredPropertySet>>>,
```

Comment:
Or either just make it live on the stylist (since I see you require it from `cascade`), but forward commands to update it to layout from script. Would that work?

Also, that would prevent the `RwLock`.